tests/unit/lms/services/oauth_http_test.py

```diff
@@ -36,24 +37,104 @@ def test_convenience_methods(self, method, svc, http_service, oauth2_token_servi
         http_service.request.side_effect = HTTPError
 
         with pytest.raises(HTTPError):
             svc.request(sentinel.method, sentinel.url)
 
+    def test_access_token_request(
+        self,
+        svc,
+        http_service,
+        OAuthTokenResponseSchema,
+        oauth_token_response_schema,
+        oauth2_token_service,
+    ):
+        oauth_token_response_schema.parse.return_value = {
+            "access_token": "access_token",
+            "refresh_token": "refresh_token",
+            "expires_in": 1234,
+        }
+
+        svc.access_token_request(
+            sentinel.token_url,
+            sentinel.redirect_uri,
+            sentinel.auth,
+            sentinel.authorization_code,
+        )
+
+        http_service.post.assert_called_once_with(
+            sentinel.token_url,
+            data={
+                "grant_type": "authorization_code",
+                "redirect_uri": sentinel.redirect_uri,
+                "code": sentinel.authorization_code,
+            },
+            auth=sentinel.auth,
+        )
+        OAuthTokenResponseSchema.assert_called_once_with(http_service.post.return_value)
+        oauth2_token_service.save.assert_called_once_with(
+            "access_token", "refresh_token", 1234
+        )
+
+    def test_access_token_request_raises_HTTPError_if_the_HTTP_request_fails(
+        self, svc, http_service
+    ):
+        http_service.post.side_effect = HTTPError
+
+        with pytest.raises(HTTPError):
+            svc.access_token_request(
+                sentinel.token_url,
+                sentinel.redirect_uri,
+                sentinel.auth,
+                sentinel.authorization_code,
+            )
+
+    def test_access_token_request_raises_ValidationError_if_the_response_is_invalid(
+        self, svc, oauth_token_response_schema
+    ):
+        oauth_token_response_schema.parse.side_effect = ValidationError({})
+
+        with pytest.raises(ValidationError):
+            svc.access_token_request(
+                sentinel.token_url,
+                sentinel.redirect_uri,
+                sentinel.auth,
+                sentinel.authorization_code,
+            )
+
+    def test_access_token_request_if_theres_no_refresh_token_or_expires_in(
+        self, svc, oauth_token_response_schema, oauth2_token_service
+    ):
+        # refresh_token and expires_in are optional fields in
+        # OAuthTokenResponseSchema so access_token_request() has to still work
+        # if they're missing from the validated data.
+        oauth_token_response_schema.parse.return_value = {
+            "access_token": "access_token"
+        }
+
+        svc.access_token_request(
+            sentinel.token_url,
+            sentinel.redirect_uri,
+            sentinel.auth,
+            sentinel.authorization_code,
+        )
+
+        oauth2_token_service.save.assert_called_once_with("access_token", None, None)
```

Comment:
Small thing, but I'd mentally group the happy cases together and the failure cases after.

Reply:
Done: https://github.com/hypothesis/lms/pull/2987/commits/dee26693a7e973d130ed226811c6831abc2a837c